src/vs/workbench/browser/positronModalDialogs/copyAsCodeModalDialog.tsx

```diff
@@ -0,0 +1,164 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (C) 2025 Posit Software, PBC. All rights reserved.
+ *  Licensed under the Elastic License 2.0. See LICENSE.txt for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+// React.
+import React, { useEffect, useState } from 'react';
+
+// Other dependencies.
+import { localize } from '../../../nls.js';
+import { ICommandService } from '../../../platform/commands/common/commands.js';
+import { IKeybindingService } from '../../../platform/keybinding/common/keybinding.js';
+import { IWorkbenchLayoutService } from '../../services/layout/browser/layoutService.js';
+import { VerticalStack } from '../positronComponents/positronModalDialog/components/verticalStack.js';
+import { PositronModalReactRenderer } from '../positronModalReactRenderer/positronModalReactRenderer.js';
+import { OKCancelModalDialog } from '../positronComponents/positronModalDialog/positronOKCancelModalDialog.js';
+import { IPositronDataExplorerInstance } from '../../services/positronDataExplorer/browser/interfaces/positronDataExplorerInstance.js';
+import { PositronDataExplorerCommandId } from '../../contrib/positronDataExplorerEditor/browser/positronDataExplorerActions.js';
+import { DropDownListBox } from '../positronComponents/dropDownListBox/dropDownListBox.js';
+import { DropDownListBoxItem } from '../positronComponents/dropDownListBox/dropDownListBoxItem.js';
+import { DropdownEntry } from './components/dropdownEntry.js';
+
+/**
+ * Shows the new folder from Git modal dialog.
+ * @param commandService The command service.
+ * @param keybindingService The keybinding service.
+ * @param layoutService The layout service.
+ * @param dataExplorerClientInstance The data explorer client instance.
+ * @returns A promise that resolves when the dialog is closed.
+ */
+export const copyAsCodeModalDialog = async (
```

Comment:
Just a small suggestion but renaming this to something like below would allow us to rename the dialog to `CopyAsCodeModalDialog `
```suggestion
export const showCopyAsCodeModalDialog = async (
```